src/lib/fetch.ts

```diff
@@ -91,11 +102,42 @@ export async function azleFetch(input: any, init?: any): Promise<any> {
             text: async () => {
                 return contents.toString();
             }
-        };
+        } as any;
     }
 
-    if (url.protocol === 'http:') {
-        // TODO http call to the management canister
+    if (url.protocol === 'https:' || url.protocol === 'http:') {
+        const body = await prepareRequestBody(init);
+        // TODO also do headers and method and everything like on the client?
+
+        const response = await azleFetch(`icp://aaaaa-aa/http_request`, {
+            body: serialize({
+                args: [
+                    {
+                        url: input,
+                        max_response_bytes: getHttpMaxResponseBytes(),
+                        method: getHttpMethod(init),
+                        headers: init?.headers ?? [],
+                        body,
+                        transform: getHttpTransform()
+                    }
+                ],
+                cycles:
+                    globalThis._azleOutgoingHttpOptionsCycles ?? 3_000_000_000n // TODO this seems to be a conservative max size
```

Comment:
could we put this in a function getCycles() like getHttpMaxRepsonseBytes()?

Reply:
It seems like we should be able to calculate the cycles based off the getHttpMaxResponseBytes() right?

Reply:
I have created the function but I have not yet tried to calculate cycles automatically, I will attempt to do that

Reply:
I have a cycles calculation, it works and is accurate according to the documentation, but ~20x more expensive than it should be according to the errors from the IC itself ($.02 per call instead of 20x less). It's just a default, I have asked in the forum to see if we can figure out why this is the case: https://forum.dfinity.org/t/a-new-price-function-for-https-outcalls/20838/21